doc/Custom-layouts.md

```diff
@@ -1,112 +1,114 @@
-
-There can be as many of these tags inside `<modmap>` as needed.
+You set the physical height of the complete keyboard in Settings as a percentage of the total height of the screen, which can be different between portrait and landscape. Unexpected Keyboard scales the "units" so all the rows fit in that alloted space. Thus, defining a shorter row gives space to the other rows.
 
-Shift and Fn modmaps also affect the clockwise circle and the roundtrip gestures.
+## Key
+The `<key />` tag defines a key on the keyboard. Its position in the sequence of keys inside `<row>`...`</row>` indicates its position in the row from left to right. It must have the following property:
+* `key0`: What the key does when tapped. Unexpected Keyboard provides a legend in the middle of the key's area.
 
-### Row
+The key tag may have additional properties as set out below.
 
-The `<row>` tag encloses one row on the keyboard. It requires no properties, and supports the following:
-
-* `height`: The height of the row. Defaults to 1, and accepts a positive floating point value.
-
-The total height of the keyboard is defined in Settings as a percentage of the total height of the screen, which can be different between portrait and landscape. The height of a row is relative to the other ones, and are scaled to keep the height of the keyboard constant.
-
-### Key
-
-The `<key />` tag defines a key on the keyboard. It requires at least one of the following properties:
-
-* `key0`: What the key should do when it is tapped.
-* `nw`, `ne`, `sw`, `se`, `w`, `e`, `n`, `s`: What the key should do when it is swiped. They are based on cardinal directions, and uses the convention that North is up. These are the new set of keywords, and should not be used with the other set of keywords.
+### Swipes
+The following optional properties define the effects of swipes:
+* `n`, `ne`, `e`, `se`, `s`, `sw`, `w`, `nw`: What the key should do when it is swiped in the direction of that compass point. ("North" means upward and "East" is to the right.)
 
  nw | n    |  ne
 :-: | :--: | :-:
  w  | key0 |  e
  sw | s    |  se
 
-* `key1` through `key8`: The older set of keywords for what the key should do when it is swiped. The directions are ordered as follows:
+* `key1` through `key8` is an older way to achieve the same effects. The directions are ordered as follows:
 
 key1 | key7 | key2
-:--: | :--: | :--:
+:-: | :--: | :-:
 key5 | key0 | key6
 key3 | key8 | key4
 
-The following properties are optionally supported:
+You can define a swipe only once with either compass-point or numeric notation. Unexpected Keyboard automatically puts a small legend in that direction from the center of the key.
+
+* `slider`: If `slider="true"`, and the key also has `w` and `e` properties, then the key tracks horizontal finger motion precisely and sends the `w` and `e` keystrokes repeatedly. In built-in layouts, this makes the space bar send left and right characters as the user slides on the space bar.
```

Comment:
The internal mechanism is more complicated than that so let's not reveal unnecessary details.
```suggestion
* `slider`: If `slider="true"`, and the key also has `w` and `e` properties, then the key tracks horizontal finger motion precisely and sends the `w` and `e` keystrokes repeatedly. In built-in layouts, this is used to implement the space bar slider.
```

Reply:
Yes, fewer details on the flow of characters is better.